crates/reward/src/lib.rs

```diff
@@ -155,7 +158,8 @@ pub mod pallet {
 
     /// Track the currencies used for rewards.
     #[pallet::storage]
-    pub type RewardCurrencies<T: Config<I>, I: 'static = ()> = StorageValue<_, BTreeSet<T::CurrencyId>, ValueQuery>;
+    pub type RewardCurrencies<T: Config<I>, I: 'static = ()> =
+        StorageMap<_, Blake2_128Concat, T::PoolId, BTreeSet<T::CurrencyId>, ValueQuery>;
```

Comment:
Why `BTreeSet`? I don't think the ordering matters, right? I think Vec is probably more efficient

Reply:
It's not so much ordering but uniqueness, harder to guarantee that a `Vec` does not contain duplicates